### DISCUSSION TITLE
Unable to start dockers component

### DISCUSSION THREAD

**Discussion by iJohnBe** [gorules/zen#154] [General]
Impossible to start the docker component : gorules/brms

I made registration to abtain a licence key for test purpose.
I setup the docker-compose with portainer as a stack (here the code)
-------------------------------------------------------------
version: '3.8'

services:
  admin:
    image: 'gorules/brms'
    container_name: gorules-brms
    ports:
      - '80:80'
    depends_on:
      - pg-database
    environment:
      DB_HOST: pg-database
      DB_PORT: 5432
      DB_USER: my-user
      DB_PASSWORD: my-password
      DB_NAME: test-db
      DB_SSL_DISABLED: true
      LICENSE_KEY: 'd563a2d5-xxxx-xxxx-xxxx-0fc4411099fb'  -- my licence key partially masked
    networks:
      - backend
      
  pg-database:
    image: 'postgres'
    container_name: pg_container
    environment:
      POSTGRES_USER: my-user
      POSTGRES_PASSWORD: my-password
      POSTGRES_DB: test-db
    ports:
      - '5432:5432'
    networks:
      - backend
            
networks:
  backend:
    external: true

-------------------------------------------------------------

container postgres start well

container gorules don't want to start (here the error log)

2024/04/29 10:47AM 30 pid=1 hostname=9e45f1e8de7b name=system msg=application starting
2024/04/29 10:47AM 30 pid=1 hostname=9e45f1e8de7b name=system msg=connecting to database on pg-database:5432
2024/04/29 10:47AM 30 pid=1 hostname=9e45f1e8de7b name=system msg=database connected on pg-database:5432
2024/04/29 10:47AM 30 pid=1 hostname=9e45f1e8de7b name=system msg=checking license
2024/04/29 10:47AM 30 pid=1 hostname=9e45f1e8de7b name=system msg=initial organisation created
2024/04/29 10:47AM 50 environment_credentials_mismatch | pid=1 hostname=9e45f1e8de7b name=system error=Bad Request statusCode=400 details={"disconnectLicense":true} msg=failed to fetch license
2024/04/29 10:47AM 50 pid=1 hostname=9e45f1e8de7b name=system err={"type":"Error","message":"incorrect license, failed to start the service","stack":"Error: incorrect license, failed to start the service\n    at /app/main.js:3:75956\n    at step (/app/main.js:3:74323)\n    at Object.next (/app/main.js:3:73625)\n    at database_asyncGeneratorStep (/app/main.js:3:72710)\n    at _next (/app/main.js:3:72942)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)"} msg=application startup error

test of : https://portal.gorules.io/api
return the following message {"message":"Route GET:/api not found","error":"Not Found","statusCode":404}

any idea of where is the error !

**Comment by stefan-gorules** [comment]
Hi @iJohnBe, I've moved the issue to discussion as this is not related to open source version.

Message above indicates that you don't have a valid license key. License keys are one time use (per database). In case you've started the product previously with a different database the issue can occur. Could you try regenerating your key?

If the issue persists, feel free to reach us at support@gorules.io.
